src/hooks/use-media-query/use-media-query.ts

```diff
@@ -2,12 +2,17 @@ import { useCallback, useEffect, useState } from 'react';
 
 import { size, SizeKey } from '~/styles/constants';
 
+import { useUserAgent } from '../use-user-agent';
+
 export function useMediaQuery(width: number): boolean;
 
 export function useMediaQuery(sizeKey: SizeKey): boolean;
 
 export default function useMediaQuery(width: number | SizeKey) {
-  const [targetReached, setTargetReached] = useState<boolean>(false);
+  const { isMobileAgent } = useUserAgent();
+  const targetWidth = typeof width === 'number' ? `${width}px` : size[width];
+  const isMobileSize = targetWidth <= size.xs && isMobileAgent;
```

Comment:
`targetWidth`가 `string` 타입일 ㄱ ㅓ 같아요 !!

그럼 이렇게 되는게 어떠신지 !!
```suggestion
  const targetWidth = typeof width === 'number' ? `${width}px` : size[width];
  const isMobileSize = (width <= size.xs || targetWidth === size.xs) && isMobileAgent;
  // 이게 맞나 ...?
```

Reply:
애매해서 SIZE라는 size의 number만 같는  상수하나 만들어서 사용하도록했습니다!